myokit/_model_api.py

```diff
@@ -4356,15 +4370,18 @@ def promote(self, state_value=0):
         if self._binding is not None:
             raise Exception(
                 'State variables cannot be bound to an external value.')
-
-        # Check state value argument
-        if isinstance(state_value, myokit.Expression):
-            if not state_value.is_literal():
-                raise myokit.NonLiteralValueError(
-                    'Expressions for state values can not contain references'
-                    ' to other variables.')
+        if not isinstance(state_value, myokit.Expression):
+            state_value = myokit.Number(state_value)
 
         model = self.model()
+
+        # check initial state value expression is constant
+        if not state_value.is_constant():
+            raise myokit.NonConstantExpressionError(
```

Comment:
I think we can delete the NonLiteralValueError then?

Reply:
done